_posts/2024-10-26-a-faster-liquidsoap-part-deux.md

```diff
@@ -9,281 +9,310 @@ _The memory part_
+requiring the programmer to do it.
 
-While this is really convenient and can help making sure that OCaml program are memory safe, this also includes another layer that is not entirely
-controllable by the application, typically when memory collection happens and how much and how long memory is allocated before it can be freed.
+While this is really convenient and can help ensure that OCaml programs are memory safe, this also includes another
+layer that is not entirely controllable by the application, typically when memory collection happens and how much and
+how long memory is allocated before it can be freed.
 
-This is even more important in the case of liquidsoap because, by nature of its streaming loop, the application constantly allocates memory under very short
-streaming cycles (typically the duration of a frame).
+This is even more important in the case of Liquidsoap because, by nature of its streaming loop, the application
+constantly allocates memory under very short streaming cycles (typically the duration of a frame).
 
-If memory collection happens too often, this results in a noticeable increase of CPU usage and, if it takes too much time, this results in an increase in memory
-usage.
+If memory collection occurs too frequently, CPU usage noticeably increases. Conversely, if it happens too infrequently,
+memory usage rises.
 
-We have [explored these issues in previous post](../2023-07-09-memory-management) and given example of parameters that can govern this trade-off. In the future,
-we would like to explore alternative memory allocation strategies to better control short term memory allocations inside the application.
+We [explored these issues in a previous post](../2023-07-09-memory-management) and gave an example of parameters that
+govern this trade-off. In the future, we would like to explore alternative memory allocation strategies to better
+control short-term memory allocations inside the application.
 
 ### How to report memory usage
 
-Now that we've seen different elements of memory usage, how can we report memory usage? It turns out that this is tricky and OS-dependent!
+Now that we've seen different elements of memory usage, how can we report memory usage? It turns out that this is tricky
+and OS-dependent!
 
-In linux, if you use `top` or similar tool, the most common measure of memory consumption is `RSS` for [_Resident set size_](https://en.wikipedia.org/wiki/Resident_set_size)
+In Linux, if you use `top` or a similar tool, the most common measure of memory consumption is `RSS` for
+[_Resident set size_](https://en.wikipedia.org/wiki/Resident_set_size).
 
-However, this number does include shared memory so, again, 10 processes using `30MB` of RSS memory does not mean that a total
-of `300MB` of the system's memory is used.
+While this number includes shared memory, it does not imply that if `10` processes each use `30MB` of RSS memory, the
```

Comment:
The logical statement is:
* Make clear that RSS does include shared memory
* Therefore, process memory is not additive

```suggestion
One important thing about RSS memory usage is that it includes shared memory. So, if `10` processes each use `30MB` of RSS memory, the
```